bundle/edu.gemini.qpt.client/src/main/java/edu/gemini/qpt/ui/view/visualizer/Visualizer.java

```diff
@@ -709,22 +721,22 @@ private Shape getElevationCurve(Function<Long, WorldCoords> coords, long start,
 			}
 
 			// Cache and return it.
-  			CachedShape cs = new CachedShape(timeAlt2XY.createTransformedShape(path), start, end);
-//			cache.put(coords, cs);
+  			cs = new CachedShapeId(timeAlt2XY.createTransformedShape(path), obsId);
+			cache.put(obsId, cs);
 			return cs.shape;
 
 		}
 	}
 
 
-	private Shape getSkyBackgroundCurve(Function<Long, WorldCoords> coords, long start, long end) {
+	private Shape getSkyBackgroundCurve(Function<Long, WorldCoords> coords, long start, long end, String obsId) {
 		synchronized (cacheLock) {
 
 			// Return the cached shape, if any.
-//			Map<WorldCoords, CachedShape> cache = skyBackgroundCurveCache;
-//			CachedShape cs = cache.get(coords);
-//			if (cs != null && cs.matches(start, end))
-//				return cs.shape;
+			Map<String, CachedShapeId> cache = skyBackgroundCurveCache;
+			CachedShapeId cs = cache.get(obsId);
+			if (cs != null && cs.matches(obsId))
```

Comment:
I think this should be set back the way it was with `start` and `end` times.  The fact that you obtained the `CachedShape` using the `obsId` as key is enough to know that it corresponds to that observation id.  However, if the shape doesn't match the same start/end time period then it shouldn't be used.  I don't think there is any need for a `CachedShapeId` object.

Otherwise this update looks good.